opennmt/utils/decoding.py

```diff
@@ -172,18 +172,21 @@ def initialize(self, batch_size, start_ids):
     initial_log_probs = tf.tile([0.] + [-float("inf")] * (self.beam_size - 1), [batch_size])
     parent_ids = tf.TensorArray(tf.int32, size=0, dynamic_size=True)
     maximum_lengths = tf.zeros([batch_size], dtype=tf.int32)
-    return start_ids, finished, initial_log_probs, (parent_ids, maximum_lengths)
+    sequence_lengths = tf.zeros([batch_size * self.beam_size], dtype=tf.int32)
```

Comment:
I think it is not necessary to use ```sequence_lengths``` for length-adaptive penalty. Because sequences meet end token will be set to ```eos_max_prob ```. Thus it will not affect the score calculation of beam searching.
https://github.com/OpenNMT/OpenNMT-tf/blob/029cc8cb81ff19c1d57422df337517e9f61eea5b/opennmt/utils/decoding.py#L290-L299

Reply:
If don't use sequence_lengths, the length penalty is invalid.
`scores /= tf.pow(((5. + tf.cast(step + 1, tf.float32)) / 6.), self.length_penalty)`
length_penalty only affects the absolute value, not relative value.